File: LanguageTutorials/DesignPatterns/src/main/java/org/bytescale/patterns/decorator/Whip.java
package org.bytescale.patterns.decorator;

public class Whip extends CondimentDecorator {
    private Beverage beverage;

    public Whip(Beverage beverage) {
        this.beverage = beverage;
    }

    @Override
    public double cost() {
        return this.beverage.cost() + .35;
    }

    @Override
    public String getDescription() {
        return this.beverage.getDescription() + ", Whip";
    }
}

File: LanguageTutorials/CSDesignPatterns/Org.Bytescale.Patterns/Decorator/Whip.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Org.Bytescale.Patterns.Decorator
{
    public class Whip : CondimentDecorator
    {
        private Beverage _beverage;
        public Whip(Beverage beverage) {
            _beverage = beverage;
        }

        public override string Description
        {
            get { return _beverage.Description + ", Whip"; }
        }

        public override double Cost
        {
            get { return _beverage.Cost + .35; }
        }
    }
}
